### PR TITLE
support new DB operations

### PR DESCRIPTION
**Pull request by follitude**
PTAL @zhangjinpeng1987  @siddontang

### CODE REVIEW

File: librocksdb_sys/crocksdb/c.cc [hunk 1]
```diff
@@ -2725,8 +2725,7 @@ void crocksdb_envoptions_destroy(crocksdb_envoptions_t* opt) { delete opt; }
 crocksdb_sstfilewriter_t* crocksdb_sstfilewriter_create(
     const crocksdb_envoptions_t* env, const crocksdb_options_t* io_options) {
   crocksdb_sstfilewriter_t* writer = new crocksdb_sstfilewriter_t;
-  writer->rep =
-      new SstFileWriter(env->rep, io_options->rep, io_options->rep.comparator);
+  writer->rep = new SstFileWriter(env->rep, io_options->rep);
```
Comment (siddontang) [inline librocksdb_sys/crocksdb/c.cc]: no need comparator now?

Comment (follitude) [inline librocksdb_sys/crocksdb/c.cc]: yes, the API has been deprecated.

**Comment by andelf** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**